packages/client/hmi-client/src/components/workflow/tera-calibration-node-ciemss.vue

```diff
@@ -288,13 +264,30 @@ const getStatus = async (simulationId: string) => {
 	}
 	console.log(`Simulation Id:${simulationId}`);
 	const runIds = [simulationId];
+
+	// open a connection for each run id and handle the messages
+	runIds.forEach((id) => {
+		eventSourceManager.openConnection(id, `/api/simulations/${id}/partial-result`);
+		eventSourceManager.setMessageHandler(id, (message) => {
+			const parsedMessage = JSON.parse(message);
+			if (parsedMessage.progress) {
+				progress.value.value = Math.round(parsedMessage.progress * 100);
+			}
+		});
+	});
+
 	poller
 		.setInterval(3000)
 		.setThreshold(300)
 		.setPollAction(async () => simulationPollAction(runIds, props.node, progress, emit));
 	console.log('Poller defined');
 	const pollerResults = await poller.start();
 
+	// closing event source connections
+	runIds.forEach((id) => {
+		eventSourceManager.closeConnection(id);
+	});
```

Comment:
You can simplify `forEach` by just calling it: `runIds.forEach(eventSourceManager.closeConnection);`

Reply:
since `closeConnection` isn't a callback function I don't believe the `id` is implicitly passed to the function.  I tried it out and it seems to be giving an error that the connection is undefined.  I have simplified the syntax a little and got rid of the `runIds.forEach( id => eventSourceManager.closeConnection(id))`

Reply:
Don't worry then.